### PR TITLE
[ios] Added destroying GL context on entering background

### PR DESCRIPTION
**Pull request by rokuz**

### CODE REVIEW

File: iphone/Maps/Common/AppInfo.mm [hunk 1]
```diff
@@ -141,4 +211,36 @@ - (NSDate *)buildDate
   return _buildDate;
 }
 
+- (NSString *)deviceName
+{
+  if (!_deviceName)
+  {
+    struct utsname systemInfo;
+    uname(&systemInfo);
+    NSString * machine =
```
Comment (syershov) [inline iphone/Maps/Common/AppInfo.mm]: use @""

File: iphone/Maps/Common/AppInfo.mm [hunk 2]
```diff
@@ -141,4 +211,36 @@ - (NSDate *)buildDate
...
+      _deviceName = machine;
+  }
+  return _deviceName;
+}
+
+- (BOOL)isMetalDriver
+{
+  struct utsname systemInfo;
+  uname(&systemInfo);
+  NSString * machine =
```
Comment (syershov) [inline iphone/Maps/Common/AppInfo.mm]: use @""

File: iphone/Maps/Common/AppInfo.mm [hunk 3]
```diff
@@ -3,13 +3,82 @@
...
+  @"iPad3,6" : @"iPad 4th gen. CDMA",
+  @"iPad4,1" : @"iPad Air WiFi",
+  @"iPad4,2" : @"iPad Air GSM",
+  @"iPad4,3" : @"iPad Air CDMA",
+  @"iPad4,4" : @"iPad Mini 2nd gen. WiFi",
+  @"iPad4,5" : @"iPad Mini 2nd gen. GSM",
+  @"iPad4,6" : @"iPad Mini 2nd gen. CDMA",
+  @"iPad5,3" : @"iPad Air 2 WiFi",
+  @"iPad5,4" : @"iPad Air 2 GSM",
+  @"iPad6,3" : @"iPad Pro (9.7 inch) WiFi",
```
Comment (syershov) [inline iphone/Maps/Common/AppInfo.mm]: move to kDeviceNamesWithiOS10MetalDriver

**Comment by darina** [comment]
LGTM

**Pull request merged**